docs/development/module/fees.md

```diff
@@ -0,0 +1,209 @@
+  mentioned, weight of each dispatch is denoted via the flexible `#[weight]` annotation. Knowing the
+  weight, it must be converted to a deductible `balance` type (typically denoted by a module that
+  implements `Currency`, `srml-balances` in substrate node). For this, each runtime must define a
+  [`WeightToFee`](/rustdocs/master/srml_transaction_payment/trait.Trait.html#associatedtype.WeightToFee)
+  type that makes the conversion. `WeightToFee` must be a struct that implements a [`Convert<Weight,
+  Balance>`](/rustdocs/master/sr_primitives/traits/trait.Convert.html).
+
+based on the above, the final fee of a dispatchable is:
+
+```
+fee =
+  base_fee +
+  len(tx) * length_fee +
+  WeightToFee(weight)
+```
+
+Customizing the above would be as simple as configuring the appropriate associated type in the
+respective module.
+
+```rust
+use sr_primitives::{traits::Convert, weights::Weight}
+// Assume this is the balance type
+type Balance = u64;
+
+// Assume we want all the weights to have a `100 + 2 * w` conversion to fees
+struct CustomWeightToFee;
+impl Convert<Weight, Balance> for CustomWeightToFee {
+    fn convert(w: Weight) -> Balance {
+        let a = Balance::from(100);
+        let b = Balance::from(2);
+        let w = Balance::from(w);
+        a + b * w
+    }
+}
+
+parameter_types! {
+    TransactionBaseFee: Balance = 10;
+    TransactionByteFee: Balance = 10;
+}
+
+impl transaction_payment::Trait {
+    TransactionBaseFee = TransactionBaseFee;
+    TransactionByteFee = TransactionByteFee;
+    WeightToFee = CustomWeightToFee;
+    // we will get to this one soon enough
+    FeeMultiplierUpdate = ();
+}
+
+```
+
+Two further questions need to be answered, given the above snippet.
+
+1. What is the purpose of `FeeMultiplierUpdate`?
+2. Knowing how we are capable of defining the conversion of weights to fees (via `WeightToFee`), the
+   question remains, how can we define and customize the weights in the first place?
+
+We will answer each question in the following sections, respectively.
+
+## Adjusting Multiplier
+The above formula gives a fee which is __logically constant through time__. Of course, the weight
+can be dynamic and based on what `WeightToFee` is defined to be, the final fee can have some degree
+of variability. As for the length fee, the inputs of the transaction could change the length and
+hence affecting the length fee. Nonetheless, these changes are independent and a _general update
+logic to the **entire fee** cannot be composed out of them trivially_. In other words, for any
+dispatchable, given the same inputs, _it will always incur the same cost_. This might not always be
+desirable. Chains might need to increase or decrease fees based on some condition.  To fulfill this
+requirement, Substrate provides:
+  - a multiplier stored in the transaction-payment module that is applied to the outcome of the
+    above formula by default (needless to say, the default value of which is _multiplication
```

Comment:
```suggestion
    above formula by default (needless to say, the default value of which is _multiplicative
```

Is this comment necessary?

Reply:
I assume yes: to emphasise that the whole thing has a switch btn.